service/internal/s3shared/xml_utils.go

```diff
@@ -16,10 +18,19 @@ type ErrorComponents struct {
 }
 
 // GetErrorResponseComponents returns the error fields from an xml error response body
-func GetErrorResponseComponents(r io.Reader) (ErrorComponents, error) {
+func GetErrorResponseComponents(r io.Reader, statusCode int, isS3service bool) (ErrorComponents, error) {
```

Comment:
Does it make sense to split these into two different functions? With the S3 version wrapping the "generic" one? That would remove the need for the boolean, and both client's being updated. S3 could have its own special version of this function